### ISSUE TITLE
Add zero-shot classification task for BLIP-2

### ISSUE THREAD

**Issue by youssefadr** [huggingface/transformers#25300]
### Feature request

I would like to add the support for the zero-shot classification task using BLIP2, computing text-image similarities with the normalized embeddings, that would be accessed from BLIP2 feature extractor. 

The idea is to enable calling the zero-shot classification pipeline using BLIP2, by implementing the `get_image_feature`and `get_text_features`methods.

I would love more guidance, if possible, on the criteria for accepting the PR.

### Motivation

This is related to the following the discussion on this issue on the hub, and the comment left by @NielsRogge here https://huggingface.co/Salesforce/blip2-opt-2.7b/discussions/3#64cbe5e487ec96aa473a1f54 .

### Your contribution

I would like to submit a PR to contribute for this feature.

**Comment by NielsRogge** [comment]
Yes so ideally you can add get_image_feature and get_text_feature  to the Blip2ForConditionalGeneration class. For that you can refer to the [original implementation ](https://github.com/salesforce/LAVIS/blob/f982acc73288408bceda2d35471a8fcf55aa04ca/lavis/models/blip2_models/blip2_qformer.py#L387).

**Comment by youssefadr** [comment]
Hello, thanks for your message, I will tackle it this week 👍

**Comment by youssefadr** [comment]
Sorry, I have been caught be in work. Will finalize the PR today!

**Comment by JhonDan1999** [comment]
> Yes so ideally you can add `get_image_feature` and `get_text_feature` to the Blip2ForConditionalGeneration class. For that you can refer to the [original implementation ](https://github.com/salesforce/LAVIS/blob/f982acc73288408bceda2d35471a8fcf55aa04ca/lavis/models/blip2_models/blip2_qformer.py#L387).

Hi I want to know if this has been done 
because I am trying to use get_image_feature but I am getting this error `AttributeError: 'Blip2ForConditionalGeneration' object has no attribute 'get_image_feature'`

and I can not use Blip2Model because I have to use `load_in_8bit` that come with Blip2ForConditionalGeneration

**Comment by NielsRogge** [comment]
Hi, no this feature hasn't been added yet.

**Comment by kirillsemenov1314** [comment]
@youssefadr hi, lmk please if help is needed here, would love to give a try to push things forward. It would actually be my first contribution, but I'm quite familiar with the BLIP2 model.
